src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -19,30 +19,40 @@ namespace ErrorCodes
     extern const int CANNOT_READ_ALL_DATA;
 }
 
+#define THROW_ARROW_NOT_OK(status)                                     \
+    do                                                                 \
+    {                                                                  \
+        if (::arrow::Status _s = (status); !_s.ok())                   \
+            throw Exception(_s.ToString(), ErrorCodes::BAD_ARGUMENTS); \
+    } while (false)
+
 ORCBlockInputFormat::ORCBlockInputFormat(ReadBuffer & in_, Block header_) : IInputFormat(std::move(header_), in_)
 {
+    prepareReader();
 }
 
 Chunk ORCBlockInputFormat::generate()
 {
```

Comment:
Do I understand correctly that previously file was read once and fully? So first call of `generate()` returned Chunk that represents the whole file, and second and others calls returned empty chunk?

Reply:
@nikitamikhaylov

**Previous Code**: 
first call `generate()` will read entire table(means entire file) into chunk and returned it and only write one huge part.
the second call `generate()` will : 
```
if(file_reader)
    return res
```
so the` res` is empty and no matter what the file size is, we only read once and write one part into disk, if the file size is huge which is will cost memory.

**Present Code**:
call  `generate()` multi times depends on the orc file has how many stripes, every stripe only read part of orc file and convert into chunk instead of converting entire file into chunk in memory by once, and then will also write multi parts into disk from orc file if it has multi stripes.

Reply:
Ok.